### ISSUE TITLE
buggy embedded openssl causes issues with config.vm.box_download_ca_cert

### ISSUE THREAD

**Issue by ericvw** [mitchellh/vagrant#5739]
The tl;dr version is that embedded `openssl` shipped with vagrant on Mac OS X is being affected by a [bug in openssl](https://rt.openssl.org/Ticket/Display.html?id=2732&user=guest&pass=guest) that fails to verify an alternate certificate path.

If you need to use a different CA bundle by using `config.vm.box_download_ca_cert` and your bundle doesn't include the 'Equifax Secure Certificate Authority' root certificate, you can run into certificate verification issues.  I came across this issue, because I need to use a CA bundle that incorporates a company root CA for HTTPS usage via a proxy.

Issues affected by this problem are probably the following: #4758 #5229 #5391 #5358.

In Mac OS X's [10.10.3 and Security Updates 2015-004](https://support.apple.com/en-us/HT204659)  update, they removed the 'Equifax Secure Certificate Authority' root CA.  So generating your own bundle from your system's keychain will result in a CA file without the Equifax root CA.

The bash script, `vagrant`, that is installed, [exports](https://github.com/mitchellh/vagrant-installers/blob/53994c49fb138b179c2b5537a8f9948644f9c26a/substrate/modules/vagrant_substrate/templates/vagrant.erb#L28) the `SSL_CERT_FILE` environment variable to be the embedded `cacert.pem`, which includes the old, weak-key (1024 bits, which is why it is being removed) [Equifax Root CA](https://github.com/mitchellh/vagrant-installers/blob/53994c49fb138b179c2b5537a8f9948644f9c26a/substrate/modules/vagrant_substrate/files/cacert.pem#L76).  However, if you provide your own CA file, which does not include the the 1024bit Equifax Root CA, one can run into certificate verification issues because of the aforementioned openssl bug.

The OpenSSL devs are aware of this issue and plan on releasing the change in 1.0.2, you can find out more details in: https://www.mail-archive.com/openssl-dev@openssl.org/msg38693.html.

In the meantime, one option is to backport the patch in the openssl that is embedded with Vagrant or wait to release another version of vagrant once openssl 1.0.2 is out.

**Comment by mitchellh** [comment]
Thanks! Next version of Vagrant is using OpenSSL 1.0.2
